src/vs/workbench/api/common/extHostNotebook.ts

```diff
@@ -372,6 +375,137 @@ export class ExtHostNotebookController implements ExtHostNotebookShape {
 		return fileStats;
 	}
 
+	/**
+	 * Search for query in all notebooks that can be deserialized by the serializer fetched by `handle`.
+	 *
+	 * @param handle used to get notebook serializer
+	 * @param filenamePattern the filename pattern to match files that can be deserialized
+	 * @param textQuery the text query to search using
+	 * @param token cancellation token
+	 * @returns `IRawClosedNotebookFileMatch` for every file. Files without matches will just have a `IRawClosedNotebookFileMatch`
+	 * 	with no `cellResults`. This allows the caller to know what was searched in already, even if it did not yield results.
+	 */
+	async $searchInNotebooks(handle: number, filenamePattern: string[], textQuery: ITextQuery, token: CancellationToken): Promise<{ results: IRawClosedNotebookFileMatch[]; limitHit: boolean }> {
+		const serializer = this._notebookSerializer.get(handle)?.serializer;
+		if (!serializer) {
+			return {
+				limitHit: false,
+				results: []
+			};
+		}
+
+		const runFileQueries = async (includes: (string)[], token: CancellationToken, textQuery: ITextQuery): Promise<URI[]> => {
+			const uris = new ResourceSet();
+			await Promise.all(includes.map(include => {
+				const query: IFileQuery = {
+					type: QueryType.File,
+					filePattern: include,
+					folderQueries: textQuery.folderQueries,
+					maxResults: textQuery.maxResults,
+				};
+				return this._extHostSearch.$doInternalFileSearchWithCustomCallback(query, token, (data) => {
+					data.forEach(e => {
+						uris.add(URI.revive(e));
```

Comment:
This method should be returning URIs, not URIComponents, since it's just called in-process

Reply:
I'm using `doInternalFileSearchWithCustomCallback` for `doInternalFileSearch` https://github.com/microsoft/vscode/blob/d2faa43f33d139236b56ada795440cefdd318e2f/src/vs/workbench/api/node/extHostSearch.ts#L111-L115

Is it still okay to have its callback return URI? For reference, the change that I think you're talking about is implemented here https://github.com/microsoft/vscode/pull/192358/commits/d2faa43f33d139236b56ada795440cefdd318e2f

Reply:
Yes, that looks correct to me. You can pass a URI to a function that takes URIComponents because URI implements that interface and will be automatically serialized when it goes over the IPC channel. But if you are calling URI.revive when passing a URI between functions in the same process, it's just wasting time